db/fetcher/fetcher.go

```diff
@@ -606,16 +621,46 @@ func (df *DocumentFetcher) fetchNext(ctx context.Context) (EncodedDocument, Exec
 			}
 		}
 
+		// Check if we have read access, for document on this collection, with the given identity.
+		if !df.passedPermissionCheck {
+			if !df.acp.HasValue() {
+				// If no acp module, then we have unrestricted access.
+				df.passedPermissionCheck = true
+			} else {
+				hasPermission, err := permission.CheckAccessOfDocOnCollectionWithACP(
+					ctx,
+					df.identity,
+					df.acp.Value(),
+					df.col,
+					acp.ReadPermission,
+					df.kv.Key.DocID,
+				)
+
+				if err != nil {
+					df.passedPermissionCheck = false
+					return nil, ExecInfo{}, err
+				}
+
+				df.passedPermissionCheck = hasPermission
+			}
+		}
+
 		// if we don't pass the filter (ran and pass)
 		// theres no point in collecting other select fields
 		// so we seek to the next doc
-		spansDone, docDone, err := df.nextKey(ctx, !df.passedFilter && df.ranFilter)
+		spansDone, docDone, err := df.nextKey(ctx, !df.passedPermissionCheck || !df.passedFilter && df.ranFilter)
```

Comment:
question: See my above comment (about ran vs passed) to Freds comment for context. Its unclear here what is going to happen if `passedPermissionCheck` is false, but not because it failed, but because it didn't need to be run (ie: No ACP). Might need to a test to determine this behavior (havent looked at the tests yet), and if so, add a specific var to differentiate between "passed" and "ran" similar to the `filter` related vars.

Reply:
If there is no ACP, passedPermissionCheck will/should always be true.

Reply:
Yupp, realized that the `acp.HasValue` basically acts the same as if there *was* an `ranPermission` var. Looks good.